pkg/cmd/pulumi-language-java/main.go

```diff
@@ -314,9 +338,42 @@ func (host *javaLanguageHost) Run(_ context.Context, req *pulumirpc.RunRequest)
 	var stdoutBuf bytes.Buffer
 	var stderrBuf bytes.Buffer
 
-	cmd.Stdout = &stdoutBuf
+	pr, pw := io.Pipe()
+
+	cmd.Stdout = pw
 	cmd.Stderr = &stderrBuf
+
+	tr := io.TeeReader(pr, &stdoutBuf)
+
 	cmd.Env = host.constructEnv(req, config, configSecretKeys)
+	go func() {
+		err := WaitForDebuggerReady(tr)
+		if err != nil {
+			logging.Errorf("failed to wait for debugger: %v", err)
+			contract.IgnoreError(cmd.Process.Kill())
+		}
+
+		// emit a debug configuration
+		debugConfig, err := structpb.NewStruct(map[string]interface{}{
+			"name":     "Pulumi: Program (Java)",
+			"type":     "java",
+			"request":  "attach",
+			"hostName": "localhost",
+			"port":     8000,
+		})
+		if err != nil {
+			logging.Errorf("failed to serialize debug configuration: %v", err)
+			contract.IgnoreError(cmd.Process.Kill())
+		}
+		_, err = engineClient.StartDebugging(ctx, &pulumirpc.StartDebuggingRequest{
+			Config:  debugConfig,
+			Message: "on port 8000",
+		})
+		if err != nil {
+			logging.Errorf("unable to start debugging: %v", err)
+			contract.IgnoreError(cmd.Process.Kill())
+		}
```

Comment:
Almost makes you want to use a `goto` :trollface:

Reply:
Tempting ;)